src/main/kotlin/io/github/cottonmc/jsonfactory/gens/DecoratedContentGenerator.kt

```diff
@@ -0,0 +1,18 @@
+package io.github.cottonmc.jsonfactory.gens
+
+import io.github.cottonmc.jsonfactory.data.Identifier
+import io.github.cottonmc.jsonfactory.output.Output
+
+/**
+ * Wrapper class, that can be used to extend the existing generators, without modifying them.
+ * */
+class DecoratedContentGenerator(
+    val superGenerator: ContentGeneratorTemplate,
```

Comment:
`delegate` would be a better name, IMO. It's used widely in Kotlin code for this pattern.

Reply:
Okey, I'll rename it. 

My logic was coming from the [decorator pattern](https://www.tutorialspoint.com/design_pattern/decorator_pattern.htm), and that is how I constructed it. The intention of that name was to communicate that it's this.

Reply:
Another name for it could be `decorated(Generator)`, but I think using super confuses it with subclassing.

Reply:
That field should probably be private.
`rootGenerator`?

Reply:
Private field + rootGenerator would work as well.

Reply:
private and renamed